Kudu.Services.Test/Diagnostics/ProcessControllerFacts.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Collections.Generic;
+using Kudu.Contracts.Settings;
+using Kudu.Contracts.Tracing;
+using Kudu.Core;
+using Kudu.Core.Diagnostics;
+using Kudu.Services.Performance;
+using Moq;
+using Xunit;
+
+namespace Kudu.Services.Test.Diagnostics
+{
+    public class ProcessControllerFacts
+    {
+        private Mock<ITracer> _tracerMock;
+        private Mock<IEnvironment> _environmentMock;
+        private Mock<IDeploymentSettingsManager> _deploymentSettingsManagerMock;
+        private ProcessController _controller;
+
+        public ProcessControllerFacts()
+        {
+            _tracerMock = new Mock<ITracer>(MockBehavior.Strict);
+            _environmentMock = new Mock<IEnvironment>(MockBehavior.Strict);
+            _deploymentSettingsManagerMock = new Mock<IDeploymentSettingsManager>(MockBehavior.Strict);
+            _controller = new ProcessController(_tracerMock.Object, _environmentMock.Object, _deploymentSettingsManagerMock.Object);
+        }
+
+        [Fact]
+        public void SetEnvironmentInfo_NotWebJob_ReturnsExpectedResults()
+        {
+            Dictionary<string, string> environmentVariables = new Dictionary<string, string>();
+            ProcessInfo processInfo = new ProcessInfo
+            {
+                EnvironmentVariables = environmentVariables
+            };
+            _controller.SetEnvironmentInfo(processInfo);
```

Comment:
These tests repro'ed the issue. Before my fix, this codepath was writing errors to ITracer